ui/README.md

```diff
@@ -1,45 +1,151 @@
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+- [How to develop](#how-to-develop)
+  - [Main technologies](#main-technologies)
+  - [Bootstrap](#bootstrap)
+    - [Global env](#global-env)
+    - [Install deps](#install-deps)
+    - [Start](#start)
+  - [Structure](#structure)
+  - [The rules we followed](#the-rules-we-followed)
+    - [TS or JS](#ts-or-js)
+    - [Styles](#styles)
+    - [Be Compact](#be-compact)
+- [License](#license)
 
 ## How to develop
 
-First, you need to provide an API server as a proxy, we will pass it into an env var which named: `REACT_APP_API_URL`.
+The following content can help you understand how to develop dashboard and its overall structure.
+
+### Main technologies
+
+<div style="display: flex; align-items: center;">
+<a href="https://www.typescriptlang.org/">
+  <img src="https://upload.wikimedia.org/wikipedia/commons/4/4c/Typescript_logo_2020.svg" height="45" alt="TypeScript" />
+</a>
+<a href="https://reactjs.org/">
+  <img src="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg" height="45" alt="React" />
+</a>
+<a href="https://redux.js.org/">
+  <img src="https://redux.js.org/img/redux.svg" height="45" alt="Redux" />
+</a>
+<a href="https://material-ui.com/" style="margin-left: 15px;">
+  <img src="https://material-ui.com/static/logo_raw.svg" height="45" alt="Material UI" />
+</a>
+</div>
+
+### Bootstrap
+
+#### Global env
+
+If you haven't installed the nodejs environment, checkout [https://nodejs.org/en/download/](https://nodejs.org/en/download/).
+
+And also, we use [Yarn 1](https://classic.yarnpkg.com/en/) as the dependency management. Maybe we will migrate to Yarn 2 in the future, but not now.
+
+#### Install deps
+
+If you just cloned a fresh Chaos Mesh repo, into the `ui` folder, run:
+
+```sh
+yarn
+```
+
+This command will install all deps the dashboard needed.
+
+Then, you need to provide an API server as a proxy, it will pass into an env var which named: `REACT_APP_API_URL`. There are three ways to get it:
+
+- **From other deployed Chaos Mesh Dashboard**
+
+  If you have Chaos Mesh deployed in the cluster, you can use the dashboard service url as the proxy.
+
+  A simple way:
+
+  ```sh
+  kubectl port-forward -n chaos-testing svc/chaos-dashboard 2333:2333
+  ```
 
-Then, into the `ui` folder, run:
+- **From local deployed Chaos Mesh Dashboard**
+
+  Same as above. Only the cluster is local. (E.g., [kind](https://kind.sigs.k8s.io/) or [minikube](https://minikube.sigs.k8s.io/))
+
+- **From local server**
+
+  There have two ways to run chaos-dashboard server in your terminal:
+
+  - `go run cmd/chaos-dashboard/main.go`
+  - `SWAGGER=1 make chaos-dashboard && ./bin/chaos-dashboard`
+
+  One is real-time, the other needs to be compiled before use. The compiled bundle the extra Swagger API HTML into the binary file.
+
+  For the other usage of `make xxx` that please refer the [Makefile](https://github.com/chaos-mesh/chaos-mesh/blob/master/Makefile).
```

Comment:
It seems to be changed to "please refer to the ...".

Reply:
Updated.~